text/0000-partial_types.md

```diff
@@ -0,0 +1,672 @@
+
+Symbol `%` percent mean percent or part of the whole thing (variable in our case).
+
+_Note_: It is highly recommended to deprecate operator `%` as a remainder function (it is still no ambiguities to write "`\s+%\s+`"), and replace it with another operator (for example: `%mod` / `%rem` / `mod` / `rem`) to not to be confused by type integrity. 
+
+
+## Traits with Integrity variants
+
+We could already write Traits with **safe** virtual functions, that consumes virtual partial types having only variants of type integrity in Trait declaration
+```rust
+// case (B1)
+pub trait Upd {
+    type UpdType;
+
+    fn summarize<%a>(&self: & %a Self) -> String;
+
+    fn update_value<%a>(&mut self : &mut %a Self, newvalue: UpdType);
+
+    fn update_sametype<%a, %b>(&mut self : &mut %a Self, &another: & %b Self);
+}
+```
+
+Unfortunately, having variants of type integrity is not enough to write **safe** implementations or other non-virtual function declarations.
+
+
+## Detailed Integrity
+
+We need detailed integrity to write non-virtual specific typed parameters in function, including trait implementation.
+
+An abstractions is added for integrity detailing, we assume that **every** variable is a `struct`-like objects (even if it is not).
+
+We need for this some new quasi-fields and some field integrity (which should be soft keywords).
+
+_Note: I do not comment types (as is needed for compiler) for colorizing purposes._
+
+_Note: I use symbol `~` as a synonym of "equivalent" word only. It is not a Rust operator._
+```rust
+// case (C1)
+let foo = 0i16;
+    foo : i16  ~  %full i16;
+    foo : %full i16  ~  %{self.*} i16;
+    foo : %{self.*} i16  ~  %{self.self} i16;
+```
+
+Where :
+ - `self` is an "access" to variable itself
+ - `.*` is an "every field" quasi-field
+ - `.self` quasi-field for primitive types, since `i16` is not a `struct` type
+
+What's about structures? Almost the same:
+```rust
+struct Point {
+    x: f64,
+    y: f64,
+    z: f64,
+    t: f64,
+    w: f64,
+}
+
+// case (C2)
+let &mut p1 : &mut Point = Point {x:1.0, y:2.0, z:3.0, t:4.0, w:5.0};
+    p1 : &mut Point  ~  &mut %full Point;
+    p1 : &mut %{self.*} Point;
+    p1 : &mut %{self.x, self.y, self.z, self.t, self.w} Point;
+    p1 : &mut %{self.{x, y, z, w}} Point;
+```
+
+Where :
+ - `.{<fld1>, <fld2>, }` is an field-set quasi-field
+
+We assume, that each field could be in one of 2 specific field-integrity - `%fit` and `%deny`. 
+
+We also must reserve as a keyword a `%miss` field-integrity for future ReExtendeded Partial Types, which allows to create **safe** self-referential types.
+
+`%fit` is default field-integrity and it means we have an access to this field and could use it as we wish. But if we try to access `%deny` field it cause a compiler error.
```

Comment:
I'm going to be honest, even after reading this multiple times, this doesn't make any sense to me at all. It doesn't help that "fit" is a word that has other meanings, and doesn't immediately strike me as being an abbreviation of "field integrity."

If the purpose of partial types is to only borrow a portion of a type, then why would you need to label what's included? Shouldn't the type be fully defined by what portion is being used?

Reply:
Ok, I've replaced "fit" to "permit" (and "integrity" to "access") for more obvious meaning.

About labels : it is a good question. 
What is really changes that? It just add additional labeling wanted to borrow fields.